app/src/main/java/org/covidwatch/android/presentation/home/HomeViewModel.kt

```diff
@@ -36,45 +41,41 @@ class HomeViewModel(
     private val _potentialRiskAction = MutableLiveData<Event<Unit>>()
     val potentialRiskAction: LiveData<Event<Unit>> get() = _potentialRiskAction
 
-    private val hasPossiblyInteractedWithInfected: LiveData<Boolean> =
-        Transformations
-            .map(contactEventDAO.allSortedByDescTimestamp) { cenList ->
-                cenList.fold(initial = false) { isInfected: Boolean, event: ContactEvent ->
-                    isInfected || event.wasPotentiallyInfectious
+    fun setup() {
+        viewModelScope.launch(Dispatchers.IO) {
+            tcnDao.allSortedByDescTimestamp()
+                .map { it.fold(false) { infected, tcn -> infected || tcn.wasPotentiallyInfectious } }
+                .collect { potentiallyInfected ->
+                    if (potentiallyInfected && !isUserTestedPositive) {
+                        _banner.postValue(
+                            Banner.Warning(
+                                R.string.contact_alert_text,
+                                BannerAction.PotentialRisk
+                            )
+                        )
+                    } else if (isUserTestedPositive) {
+                        _banner.postValue(
+                            Banner.Warning(
+                                R.string.reported_alert_text,
+                                BannerAction.PotentialRisk
+                            )
+                        )
+                    }
```

Comment:
what  if we do that in `init {}` block? because we call `setup()` every time we open `HomeFragment`?

Reply:
`init` block is not good for testing.
The problems with calling every time can be resolved with adding another method that we would call in `onCreate` of the Fragment.